Options.py

```diff
@@ -1315,6 +1315,7 @@ def as_dict(
         will be returned as a sorted list.
         """
         assert option_names, "options.as_dict() was used without any option names."
+        assert len(option_names) < len(self.__class__.type_hints), "Specify only options you need."
```

Comment:
Are we assuming that slot data is the only use of this function?
That doesn't seem like a very safe assumption.

We could check the stack when the function is called to see if it's making slot data, and only apply the restriction in that case...